ush/python/pygw/src/pygw/configuration.py

```diff
@@ -137,16 +121,59 @@ def _get_shell_env(scripts: List) -> Dict[str, Any]:
             varbls[entry[0:iequal]] = entry[iequal + 1:]
         return varbls
 
-    @staticmethod
-    def _cast_or_not(type, value):
+
+def cast_strdict_as_dtypedict(ctx: Dict[str, str]) -> Dict[str, Any]:
+    """
+    Environment variables are typically stored as str
+    This method attempts to translate those into datatypes
+    Parameters
+    ----------
+    ctx : dict
+          dictionary with values as str
+    Returns
+    -------
+    varbles : dict
+              dictionary with values as datatypes
+    """
+    varbles = AttrDict()
+    for key, value in ctx.items():
+        varbles[key] = cast_as_dtype(value)
+    return varbles
+
+
+def cast_as_dtype(string: str) -> Union[str, int, float, bool, Any]:
+    """
+    Cast a value into known datatype
+    Parameters
+    ----------
+    string: str
+    Returns
+    -------
+    value : str or int or float or datetime
+            default: str
+    """
+    TRUTHS = ['y', 'yes', 't', 'true', '.t.', '.true.']
+    BOOLS = ['n', 'no', 'f', 'false', '.f.', '.false.'] + TRUTHS
+    BOOLS = [x.upper() for x in BOOLS] + BOOLS + ['Yes', 'No', 'True', 'False']
+
+    def _cast_or_not(type: Any, string: str):
         try:
-            return type(value)
+            return type(string)
         except ValueError:
-            return value
+            return string
 
-    @staticmethod
-    def _true_or_not(value):
+    def _true_or_not(string: str):
         try:
-            return value.lower() in Configuration.TRUTHS
+            return string.lower() in TRUTHS
```

Comment:
We're kinda treating the lists inconsistently. `TRUTHS` gets a single case format and all inputs are converted to lower for comparison, but `BOOLS` gets a number of allowed capitalizations (upper, lower, title) and has to match one of them. Personally prefer the former, so typos like `TRue` would match, but fine with either choice.

Reply:
I understand that, but trying to account and anticipate for "user-error" and correct it is not usually considered safe practice (Don't remember where I read that).  Instead, I would be inclined to add an exception stating that `FAlse` or `trUe` is not an acceptable value.

Also note how these (bools) are cast.
The `if` statement checks if the `string` is in `BOOLS` (Upper and lowercase, as well as the Titlecase).  If the condition is satisfied, the `string` is converted to `lower` to match if it is `True` if the `string` is indeed `True`.

To extend `BOOLS` to account for the typos, would mean to extend `BOOLS` to:
```
BOOLS +=  ['yEs', 'yES', 'yeS', 'YeS']
```
And many more for `True` and `False`, `.TRUE.', `.FALSE'.

Reply:
Still feels weird that if you call `_true_or_not()` directly on `"TRue"`, you'll get `True`, but if you check whether it is a boolean by looking for it in `BOOLS`, it isn't. Even if the current implementation only calls `_true_or_not()` after first checking `BOOLS`.

Reply:
Note I have just moved the methods out of the `class Configuration` to make use of them outside.
Feel free to open a PR to improve/expand the use of typo-ed booleans.